From 92641fe68ccf880e9187b854fcf92427bd498a3e Mon Sep 17 00:00:00 2001
From: Tomas Glozar <tglozar@gmail.com>
Date: Sat, 15 Nov 2025 21:52:11 +0100
Subject: [PATCH 39/39] sysdeps/ia64: Drop unneeded cmp.ne.and from strcmp

It is only needed to compare one of val1, val2 against zero, because
whenever val1 != 0 && val2 == 0, then also val1 != val2.

Drop the second comparison, and reduce the loop to two bundles,
leading to improved performance.

Signed-off-by: Tomas Glozar <tglozar@gmail.com>
---
 sysdeps/ia64/strcmp.S | 1 -
 1 file changed, 1 deletion(-)

diff --git a/sysdeps/ia64/strcmp.S b/sysdeps/ia64/strcmp.S
index 0a86dd1c5a..1fa4fb0f5c 100644
--- a/sysdeps/ia64/strcmp.S
+++ b/sysdeps/ia64/strcmp.S
@@ -43,7 +43,6 @@ ENTRY(strcmp)
 	cmp.eq	p6, p0 = r0, r0		// set p6
 	;;
 	cmp.ne.and p6, p0 = val1, r0
-	cmp.ne.and p6, p0 = val2, r0
 	cmp.eq.and p6, p0 = val1, val2
 (p6)	br.cond.sptk .loop
 	sub	ret0 = val1, val2
-- 
2.51.0

